minetest/minetest
# Fix camera to follow player eyes when player is attached

- Goal of the PR
Make the player camera follow the player's eyes independently on attached offset and rotation and parent object rotation.
- How does the PR work?
Change the camera position setting.
- Does it resolve any reported issue?
This probably resolves bug issue #10101 and more.
- Does this relate to a goal in [the roadmap](https://github.com/minetest/minetest/blob/master/doc/direction.md)?
I have no idea.

## To do

This PR is a Ready to Review

## How to test

Run devtest game, add `testentities:attach` entity to the scene, and right-click to attach yourself to it.
Control keys are up, down, left, right, jump, sneak, and aux1. `aux1` changes the mode of control.
To check `get_look_dir`, take the `testlookdir:fire` item and use the dig/place buttons to the fire entity in the direction of `look_dir` get from `get_look_dir` or calculated from `get_look_vertical` and `get_look_horizontal`.

You can detach this branch HEAD to one commit before commit "Fix camera position and rotation." to see how it works without a camera fix.

See before :
https://streamable.com/ih3lb5

And after:
https://streamable.com/tvdztt
